app/models/host/managed.rb

```diff
@@ -961,10 +961,14 @@ def assign_hostgroup_attributes(attrs = [])
     attrs.each do |attr|
       next if send(attr).to_i == -1
       value = hostgroup.send("inherited_#{attr}")
-      self.send("#{attr}=", value) unless send(attr).present?
+      assign_hostgroup_attribute attr, value
     end
   end
 
+  def assign_hostgroup_attribute(attr, value)
+    self.send("#{attr}=", value) if new_record? || !send(attr).present?
```

Comment:
Ah I see it now, you want to preserve the options  for existing hosts. Can you move this check to where this method is called? `assign_hostgroup_attribute(attr, value) if new_record?`